crates/services/txpool_v2/src/pool.rs

```diff
@@ -0,0 +1,196 @@
+                    let blob_id = checked_tx.transaction().blob_id();
+                    if latest_view
+                        .blob_exist(blob_id)
+                        .map_err(|e| Error::Database(format!("{:?}", e)))?
+                    {
+                        return Err(Error::NotInsertedBlobIdAlreadyTaken(*blob_id))
+                    }
+                }
+                let collisions = self
+                    .collision_manager
+                    .collect_colliding_transactions(&tx, &self.storage)?;
+                let dependencies = self.storage.collect_dependencies_transactions(
+                    &tx,
+                    collisions.reasons,
+                    &latest_view,
+                    self.config.utxo_validation,
+                )?;
+                let has_dependencies = !dependencies.is_empty();
+                let (storage_id, removed_transactions) = self.storage.store_transaction(
+                    tx,
+                    dependencies,
+                    collisions.colliding_txs,
+                )?;
+                #[cfg(test)]
+                {
+                    self.tx_id_to_storage_id.insert(tx_id, storage_id);
+                }
+                // No dependencies directly in the graph and the sorted transactions
+                if !has_dependencies {
+                    self.selection_algorithm
+                        .new_executable_transactions(vec![storage_id], &self.storage)?;
+                }
+                let tx = Storage::get(&self.storage, &storage_id)?;
+                let result = removed_transactions
+                    .into_iter()
+                    .map(|tx| {
+                        self.collision_manager.on_removed_transaction(&tx)?;
+                        self.selection_algorithm.on_removed_transaction(&tx)?;
+                        #[cfg(test)]
+                        {
+                            self.tx_id_to_storage_id.remove(&tx.id());
+                        }
+                        Ok(tx)
+                    })
+                    .collect();
+                self.collision_manager
+                    .on_stored_transaction(&tx.transaction, storage_id)?;
+                result
+            })
+            .collect())
+    }
+
+    // TODO: Use block space also (https://github.com/FuelLabs/fuel-core/issues/2133)
+    /// Extract transactions for a block.
+    /// Returns a list of transactions that were selected for the block
+    /// based on the constraints given in the configuration and the selection algorithm used.
+    pub fn extract_transactions_for_block(
+        &mut self,
+    ) -> Result<Vec<PoolTransaction>, Error> {
+        self.selection_algorithm
+            .gather_best_txs(
+                Constraints {
+                    max_gas: self.config.max_block_gas,
+                },
+                &self.storage,
+            )?
+            .into_iter()
+            .map(|storage_id| {
```

Comment:
I feel like this closure could be moved to a helper. `self.remove_transaction` or something.

Reply:
Done.